cabal-testsuite/src/Test/Cabal/Prelude.hs

```diff
@@ -612,9 +614,12 @@ withRepoNoUpdate repo_dir m = do
     withReaderT (\env' -> env' { testHaveRepo = True }) m
     -- TODO: Arguably should undo everything when we're done...
   where
-    repoUri env ="file+noindex://" ++ (if isWindows
-                                        then joinDrive "//./" . asPosixPath
-                                        else id) (testRepoDir env)
+    repoUri env ="file+noindex:" ++ (if isWindows
+                                        then map (\x -> if x == Windows.pathSeparator
+                                                        then Posix.pathSeparator
+                                                        else x
+                                                 )
+                                        else ("//" ++)) (testRepoDir env)
```

Comment:
`cabal-testsuite` does not depend on `cabal-install` so I had to inline this transformation.